docs/advanced-querying-techniques/ctes.md

```diff
@@ -0,0 +1,398 @@
+This code installs JupySQL, and DuckDB in your environment. We will be using these moving forward.
+
+```{code-cell} ipython3
+%pip install jupysql --upgrade jupysql-plugin --upgrade duckdb-engine --quiet
+```
+
+We continue to work with the Bank and Marketing data set.
+
+```{important}
+Source: UCI Machine Learning Repository
+
+URL: https://archive-beta.ics.uci.edu/dataset/222/bank+marketing
+
+Data Citation
+
+Moro,S., Rita,P., and Cortez,P.. (2012). Bank Marketing. UCI Machine Learning Repository. https://doi.org/10.24432/C5K306.
+```
+
+We can use the following function to extract the downloaded data from the UCI repository.
+
+```{code-cell} ipython3
+import sys
+
+sys.path.insert(0, "../../")
+import banking  # noqa: E402
+
+_ = banking.BankingData("https://tinyurl.com/jb-bank", "bank")
+_.extract_to_csv()
+```
+
+Initialize a DuckDB Instance
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%reload_ext sql
+# Initiating a DuckDB database named 'bank.duck.db' to run our SQL queries on
+%sql duckdb:///bank.duck.db
+```
+
+Load the data
+
+```{code-cell} ipython3
+%%sql
+CREATE OR REPLACE TABLE bank AS
+FROM read_csv_auto('bank_cleaned.csv', header=True, sep=',')
+```
+
+We confirm the table was loaded
+
+```{code-cell} ipython3
+%sqlcmd tables
+```
+
+We can use [JupySQL's Table Explorer](https://jupysql.ploomber.io/en/latest/user-guide/table_explorer.html) to take a quick look at the table.
+
+```{code-cell} ipython3
+%sqlcmd explore --table bank
+```
+
+## Simple CTE
+
+Let's create a CTE that finds the average balance for each job type.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+)
+SELECT * 
+FROM Job_Avg_Balance;
+```
+
+In this example, the `WITH` keyword indicates the start of the CTE, which we call `Job_Avg_Balance`.
+
+Within the `Job_Avg_Balance` CTE, we form a query that returns the average balance for each job by computing the average of `balance` and grouping by `job`. 
+
+After closing the `WITH` statement, we then select all columns from the `Job_Avg_Balance` CTE. This returns two columns: `job` and `AverageBalance`. The final `SELECT` statement then retrieves the data from the CTE.
+
+## Multiple CTEs
+
+You can use multiple CTEs in a single query. Let's find the average balance per job type and average campaign per job type.
+
+```{code-cell} ipython3
+%%sql
+WITH Job_Avg_Balance AS (
+    SELECT job, AVG(balance) AS AverageBalance
+    FROM bank
+    GROUP BY job
+),
+Job_Avg_Campaign AS (
+    SELECT job, AVG(campaign) AS AverageCampaign
+    FROM bank
+    GROUP BY job
+)
+SELECT * 
+FROM Job_Avg_Balance, Job_Avg_Campaign 
+WHERE Job_Avg_Balance.job = Job_Avg_Campaign.job;
+```
+
+In this example, the first CTE is the same as in the previous example. The second CTE `Job_Avg_Campaign` returns two columns: `job_1` and `AverageCampaign`. The final SELECT statement retrieves data from both CTEs. Notice that the entries.
```

Comment:
`Notice that the entries.` -> This sentence looks incomplete

Reply:
Fixed